batched/dense/unit_test/Test_Batched_SerialTbsv.hpp

```diff
@@ -0,0 +1,326 @@
+          h_x_ref(ib, 0)         = 1.0;
+          h_x_ref(ib, 1)         = 0.0;
+          h_x_ref(ib, 2)         = 0.0;
+          h_x1_ref(ib, 0)        = 1.0;
+          h_x1_ref(ib, incx)     = 0.0;
+          h_x1_ref(ib, 2 * incx) = 0.0;
+        } else {
+          h_x_ref(ib, 0)         = 1.0;
+          h_x_ref(ib, 1)         = 0.0;
+          h_x_ref(ib, 2)         = 0.0;
+          h_x1_ref(ib, 0)        = 1.0;
+          h_x1_ref(ib, incx)     = 0.0;
+          h_x1_ref(ib, 2 * incx) = 0.0;
+        }
+      }
+    } else {
+      if constexpr (std::is_same_v<typename ParamTagType::trans,
+                                   Trans::NoTranspose>) {
+        if constexpr (std::is_same_v<typename ParamTagType::diag,
+                                     Diag::NonUnit>) {
+          h_x_ref(ib, 0)         = 1.0;
+          h_x_ref(ib, 1)         = -1.0 / 2.0;
+          h_x_ref(ib, 2)         = -1.0 / 6.0;
+          h_x1_ref(ib, 0)        = 1.0;
+          h_x1_ref(ib, incx)     = -1.0 / 2.0;
+          h_x1_ref(ib, 2 * incx) = -1.0 / 6.0;
+        } else {
+          h_x_ref(ib, 0)         = 1.0;
+          h_x_ref(ib, 1)         = -1.0;
+          h_x_ref(ib, 2)         = 1.0;
+          h_x1_ref(ib, 0)        = 1.0;
+          h_x1_ref(ib, incx)     = -1.0;
+          h_x1_ref(ib, 2 * incx) = 1.0;
+        }
+      } else {
+        if constexpr (std::is_same_v<typename ParamTagType::diag,
+                                     Diag::NonUnit>) {
+          h_x_ref(ib, 0)         = 0.0;
+          h_x_ref(ib, 1)         = 0.0;
+          h_x_ref(ib, 2)         = 1.0 / 3.0;
+          h_x1_ref(ib, 0)        = 0.0;
+          h_x1_ref(ib, incx)     = 0.0;
+          h_x1_ref(ib, 2 * incx) = 1.0 / 3.0;
+        } else {
+          h_x_ref(ib, 0)         = 2.0;
+          h_x_ref(ib, 1)         = -2.0;
+          h_x_ref(ib, 2)         = 1.0;
+          h_x1_ref(ib, 0)        = 2.0;
+          h_x1_ref(ib, incx)     = -2.0;
+          h_x1_ref(ib, 2 * incx) = 1.0;
+        }
+      }
+    }
+  }
+
+  Kokkos::fence();
+
+  Kokkos::deep_copy(ref, h_ref);
+  Kokkos::deep_copy(x0, h_x0);
+  Kokkos::deep_copy(x1, h_x1);
+  Kokkos::deep_copy(x_ref, h_x_ref);
+  Kokkos::deep_copy(x1_ref, h_x1_ref);
+
+  // Create triangluar or banded matrix
+  create_banded_triangular_matrix<View3DType, View3DType,
+                                  typename ParamTagType::uplo>(ref, A, k,
+                                                               false);
+  create_banded_triangular_matrix<View3DType, View3DType,
+                                  typename ParamTagType::uplo>(ref, Ab, k,
+                                                               true);
+
+  // tbsv
+  Functor_BatchedSerialTbsv<DeviceType, View3DType, View2DType, ParamTagType,
+                            AlgoTagType>(Ab, x0, k, 1)
+      .run();
+
+  // tbsv with incx == 2
+  Functor_BatchedSerialTbsv<DeviceType, View3DType, View2DType, ParamTagType,
```

Comment:
Users should not need to use `incx`, that information usually is carried by the layout of the `Kokkos::View` x1

Reply:
I am not quite sure about this point. If I understand correctly, `incx` can be a negative number, so it cannot be computed from View extents only.